sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReactorAmqpConnection.java

```diff
@@ -197,6 +198,35 @@ public ServiceBusReactorAmqpConnection(String connectionId, ConnectionOptions co
             });
     }
 
+    @Override
+    public Mono<TransactionManager> getTransactionManager() {
+        final String linkName = "coordinator";
+        if (isDisposed()) {
+            return Mono.error(logger.logExceptionAsError(new IllegalStateException(String.format(
+                "connectionId[%s]: Connection is disposed. Cannot get transaction coordinator instance for '%s'",
+                connectionId, linkName))));
+        }
+
+
+        final TransactionManager existing = transactionManagers.get(linkName);
+        if (existing != null) {
+            return Mono.just(existing);
+        }
+
+        return getReactorConnection().then(
+            Mono.defer(() -> Mono.just(transactionManagers.compute(linkName, (key, current) -> {
```

Comment:
Why mono.defer?

Reply:
removed it, not using transactionmanager